src/pages/index.js

```diff
@@ -164,39 +179,93 @@ profileEditButton.addEventListener("click", (evt) => {
 
 editForm.addEventListener("submit", handleEditFormSubmit);
 
-initialCards.forEach((card) => {
-  const cardElement = getCardElement(card);
-  cardsList.prepend(cardElement);
-});
-
 // New Post Modal elements
 const newPostButton = document.querySelector(".profile__add-btn");
 const newPostModal = document.querySelector("#add-card-modal");
-const newPostForm = document.forms["add-card-form"];
-
-const newPostLinkInput = newPostModal.querySelector("#add-card-link-input");
-const newPostCaptionInput = newPostModal.querySelector(
+const newPostForm = document.querySelector("#add-card-form");
+const newPostLinkInput = newPostForm.querySelector("#add-card-link-input");
+const newPostCaptionInput = newPostForm.querySelector(
   "#add-card-caption-input"
 );
 
 // For New Post Modal functions
-newPostButton.addEventListener("click", (evt) => {
+newPostButton.addEventListener("click", () => {
   openModal(newPostModal);
 });
 
 newPostForm.addEventListener("submit", handleNewFormSubmit);
 
 function handleNewFormSubmit(evt) {
   evt.preventDefault();
-  const inputValues = {
-    name: newPostCaptionInput.value,
-    link: newPostLinkInput.value,
-  };
-  const cardEl = getCardElement(inputValues);
-  cardsList.prepend(cardEl);
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .addNewCard({
+      name: newPostCaptionInput.value,
+      link: newPostLinkInput.value,
+    })
+    .then((data) => {
+      const cardEl = getCardElement(data);
+      cardsList.prepend(cardEl);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, true);
+    });
+
   evt.target.reset();
   disableButton(newPostSubmitBtn, settings);
   closeModal(newPostModal);
```

Comment:
```js


  evt.target.reset();
  disableButton(newPostSubmitBtn, settings);
  closeModal(newPostModal);
```

These 3 lines should be in a `.then` block in line 210

Reply:
You need to **reset** the form only  in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.

Reply:
You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.